File: algorithms/475. Heaters/solution.py
class Solution(object):
    def findRadius(self, houses, heaters):
        """
        :type houses: List[int]
        :type heaters: List[int]
        :rtype: int
        """
        heaters = sorted(heaters) + [sys.maxsize]
        i = r = 0
        for x in sorted(houses):
            while x >= (heaters[i] + heaters[i+1]) / 2.0:
                i += 1
            r = max(r, abs(heaters[i] - x))
        return r

# 27 / 27 test cases passed.
# Status: Accepted
# Runtime: 152 ms

File: algorithms/475. Heaters/solution.cc
class Solution {
public:
    int findRadius(vector<int>& houses, vector<int>& heaters) {
        sort(houses.begin(), houses.end());
        sort(heaters.begin(), heaters.end());
        int radius = 0, i = 0;
        for ( int house : houses ) {
            while ( i < heaters.size() - 1 && heaters[i] + heaters[i+1] < 2 * house ) {
                ++i;
            }
            radius = max(radius, abs(house - heaters[i]));
        }
        return radius;
    }
};
